community/contributing/back-end-code.md

```diff
@@ -74,3 +74,21 @@ For example, these names are much less likely to result in errors:
 public long getTimeoutMillis();
 public long getTimeoutSeconds();
 ```
+
+## Exceptions
+
+Please select, or create, appropriate exception types rather than throwing overly general things such as `RuntimeException`.
+
+Exception types you create should extend `RuntimeException` (directly or indirectly).
+Please try to include descriptive information in exception messages, especially for errors that will be surfaced to the user.
+
+## Testing
+
```

Comment:
Do you have strong feelings on test format? Naming of tests in `speck`/`spock`? Grouping? Those opinions would be helpful to know.